c/xen_common.c

```diff
@@ -885,22 +885,27 @@ static int count_children(xmlNode *n, const char *name)
 static void destring(xen_session *s, xmlChar *name, const abstract_type *type,
                      void *value)
 {
-    switch (type->typename)
```

Comment:
Why do we replace the switch?
Switch seems more efficient and this function is a critical path.